### PR TITLE
feat: Allow advanced routing patterns with RabbitMQ plugin

### PR DESCRIPTION
**Pull request by luislhl**
### Acceptance Criteria

The RabbitMQ plugin should allow usage of advanced routing patterns in RabbitMQ.

For this, it should accept 2 new arguments: the exchange name and routing key, and validate whether either an exchange or a queue is provided, not both.

Previously it only supported sending to a single queue, which prevented developers from using it to send messages to Exchanges, which are needed for some more advanced routing strategies (see https://www.rabbitmq.com/tutorials/amqp-concepts#exchanges)

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/plugins/hathor_rabbitmq.js [hunk 1]
```diff
@@ -33,12 +33,40 @@ export function getSettings() {
...
+  }
+
+  return { url, queue, exchange, routingKey };
 }
 
 export function eventHandlerFactory(channel, settings) {
+  if (settings.exchange !== undefined) {
+    return data => {
+      // Check https://amqp-node.github.io/amqplib/channel_api.html#channel_publish
+      channel.publish(settings.exchange, settings.routingKey, Buffer.from(JSON.stringify(data)));
```
[the remark this answers is not in the document]
Comment (luislhl) [reply src/plugins/hathor_rabbitmq.js:64]: Yes, I'll rebase this PR with master and change this

[the remark this answers is not in the document]
Comment (luislhl) [reply src/plugins/hathor_rabbitmq.js]: Fixed in https://github.com/HathorNetwork/hathor-wallet-headless/pull/531/commits/2774a74b70013408bf26077f2e8a9a942dad12f3

**Review by r4mmer — DISMISSED** [review]

File: __tests__/plugins/rabbitmq_plugin.test.js [hunk 2]
```diff
@@ -8,32 +8,110 @@
...
+
+  it('should throw an error if exchange is provided without routing key', () => {
+    const oldArgs = process.argv;
+    process.argv = [
+      'node', // not used but a value is required at this index
+      'a_script_file.js', // not used but a value is required at this index
+      '--plugin_rabbitmq_url', 'test-url',
+      '--plugin_rabbitmq_exchange', 'test-exchange',
+    ];
+    expect(() => getSettings()).toThrow('You must provide a RabbitMQ routing key if you provide exchange. A blank routing key is acceptable though.');
```
Comment (tuliomir) [inline __tests__/plugins/rabbitmq_plugin.test.js:54]: suggestion(non-blocking): Not related to this line, but I think we should have a test for this edge case. An exchange with a blank routing key.

This will ensure the command line parser is working as expected. This doesn't need to be implemented in this PR, however, as it's just an improvement upon this working set of tests.

File: __tests__/plugins/rabbitmq_plugin.test.js [hunk 3]
```diff
@@ -8,32 +8,110 @@
...
+    process.argv = oldArgs;
+  });
+
+  it('should return the settings if everything is correct', () => {
```
Comment (tuliomir) [inline __tests__/plugins/rabbitmq_plugin.test.js:58]: suggestion(non-blocking): improve wording to match the test below
```suggestion
  it('should return the settings if everything is correct with queue configuration', () => {
```

File: src/plugins/hathor_rabbitmq.js [hunk 4]
```diff
@@ -36,12 +36,44 @@ export function getSettings() {
...
+      channel.publish(
+        settings.exchange,
+        settings.routingKey,
+        Buffer.from(bigIntUtils.JSONBigInt.stringify(data))
+      );
+    };
+  }
+
   return data => {
+    // Check https://amqp-node.github.io/amqplib/channel_api.html#channel_sendToQueue
```
Comment (tuliomir) [inline src/plugins/hathor_rabbitmq.js:76]: :star: Nice to have the urls to the docs in the code!

**Review by tuliomir — DISMISSED** [review]

File: __tests__/plugins/rabbitmq_plugin.test.js (hunk 2 above)
Comment (luislhl) [reply __tests__/plugins/rabbitmq_plugin.test.js:54]: Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/531/commits/0301b03f9fb11d0c2e28adc9b539e9d5a93218cc

File: __tests__/plugins/rabbitmq_plugin.test.js (hunk 3 above)
Comment (luislhl) [reply __tests__/plugins/rabbitmq_plugin.test.js]: Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/531/commits/0301b03f9fb11d0c2e28adc9b539e9d5a93218cc
